src/components/MapView.js

```diff
@@ -0,0 +1,427 @@
+    const people = items[state][district] ? items[state][district] : [];
+    if (people.length) {
+      const incumbent = people.filter(person => person.incumbent === true)[0] || false;
+      if (incumbent) {
+        tooltip += `<div>Incumbent <strong>${incumbent.displayName}</strong>${takenThePledge(incumbent)}</div>`;
+      }
+      people.filter(person => person.incumbent === false).forEach((person) => {
+        tooltip += `<div> Candidate <strong>${person.displayName}</strong>${takenThePledge(person)}</div>`;
+      });
+    } else {
+      tooltip += '<div>No one in this district has signed the pledge yet.</div>';
+    }
+    return tooltip;
+  }
+
+  addPopups(layer) {
+    const { map } = this;
+    const popup = new mapboxgl.Popup({
+      closeButton: false,
+      closeOnClick: false,
+    });
+    const { items } = this.props;
+
+    map.on('mousemove', (e) => {
+      const features = map.queryRenderedFeatures(e.point, { layers: [layer] });
+      // Change the cursor style as a UI indicator.
+      map.getCanvas().style.cursor = (features.length) ? 'pointer' : '';
+
+      if (features.length) {
+        const feature = features[0];
+        // ABR:"ME",
+        // GEOID:"2302"
+        // NAME:"Maine"
+        // STATEFP:"23"
+        const { properties } = feature;
+        const stateAbr = properties.ABR;
+        let tooltip;
+        if (map.metadata.level === 'districts') {
+          if (!items[stateAbr]) {
+            return undefined;
+          }
+          tooltip = this.showDistrictTooltip(stateAbr, Number(properties.GEOID.substring(2)));
+        } else {
+          tooltip = this.showStateTooltip(stateAbr);
+        }
+
+        return popup.setLngLat(e.lngLat)
+          .setHTML(tooltip)
+          .addTo(map);
+      }
+      return undefined;
+    });
+  }
+
+  focusMap(bb) {
+    if (!bb) {
+      return;
+    }
+    const height = window.innerHeight;
+    const width = window.innerWidth;
+    const view = geoViewport.viewport(bb, [width / 2, height / 2]);
+    if (view.zoom < 2.5) {
+      view.zoom = 2.5;
+    } else {
+      view.zoom -= 0.5;
+    }
+    this.map.flyTo(view);
+  }
+
+  insetOnClickEvent(e) {
+    const dataBounds = e.target.parentNode.parentNode.getAttribute('data-bounds').split(',');
+    const boundsOne = [Number(dataBounds[0]), Number(dataBounds[1])];
+    const boundsTwo = [Number(dataBounds[2]), Number(dataBounds[3])];
+    const bounds = boundsOne.concat(boundsTwo);
+    this.map.fitBounds(bounds);
+  }
+
+  removeHighlights() {
+    this.map.setLayoutProperty('selected-fill', 'visibility', 'none');
+    this.map.setLayoutProperty('selected-border', 'visibility', 'none');
+  }
+
+  handleReset() {
+    this.removeHighlights();
+    this.props.resetSelections();
+  }
+  // Creates the button in our zoom controls to go to the national view
+  makeZoomToNationalButton() {
+    document.querySelector('.mapboxgl-ctrl-compass').remove();
```

Comment:
Can't this be done while initing the map?

Reply:
you mean in the styles? It is being called in the initializeMap function. This is Dylan's code, but it seemed fine